unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/Editor/Procedural Skybox/SkyboxEditor.cs

```diff
@@ -0,0 +1,1344 @@
+            if (configurations == null || configurations.Count < 1)
+            {
+                selectedConfiguration = AddNewConfiguration("Generic Skybox");
+
+                configurations = new List<SkyboxConfiguration>();
+                configurations.Add(selectedConfiguration);
+            }
+
+            if (selectedConfiguration == null)
+            {
+                selectedConfiguration = configurations[0];
+            }
+
+            for (int i = 0; i < configurations.Count; i++)
+            {
+                configurationNames.Add(configurations[i].skyboxID);
+                if (selectedConfiguration == configurations[i])
+                {
+                    selectedConfigurationIndex = i;
+                }
+            }
+
+            UpdateSlotsID();
+
+            InitializeMaterial();
+
+            //isPaused = true;
+
+            if (!Application.isPlaying)
+            {
+                isPaused = true;
+            }
+
+
+        }
+
+        /// <summary>
+        /// Update slot ID for old configurations. (Backward comptibility)
+        /// </summary>
+        private void UpdateSlotsID()
+        {
+            for (int i = 0; i < selectedConfiguration.slots.Count; i++)
+            {
+                selectedConfiguration.slots[i].UpdateSlotsID(i);
+            }
+        }
+
+        private void RenderTimePanel()
+        {
+
+            GUILayout.Label("Preview", EditorStyles.boldLabel);
+
+            GUILayout.BeginHorizontal();
+            GUILayout.Label("Time : " + timeOfTheDay, EditorStyles.label, GUILayout.MinWidth(100));
+            timeOfTheDay = EditorGUILayout.Slider(timeOfTheDay, 0.01f, 24.00f, GUILayout.MinWidth(100), GUILayout.MaxWidth(300));
+            EditorGUILayout.LabelField((GetNormalizedDayTime() * 100).ToString("f2"), GUILayout.MaxWidth(50));
+
+            if (isPaused)
+            {
+                if (GUILayout.Button("Play"))
+                {
+                    ResumeTime();
+                }
+            }
+            else
+            {
+                if (GUILayout.Button("Pause"))
+                {
+                    PauseTime();
+                }
+            }
+
+            GUILayout.EndHorizontal();
+        }
+
+        void ResumeTime()
+        {
+            isPaused = false;
+            if (Application.isPlaying)
+            {
+                if (SkyboxController.i != null)
+                {
+                    SkyboxController.i.ResumeTime(true, timeOfTheDay);
+                }
+            }
+        }
+
+        void PauseTime()
+        {
+            isPaused = true;
+            if (Application.isPlaying)
+            {
+                if (SkyboxController.i != null)
+                {
+                    SkyboxController.i.PauseTime();
+                }
+            }
+        }
+
+        void RenderBackgroundColorLayer()
+        {
+
+            //GUILayout.Label("Background Layer", EditorStyles.boldLabel);
```

Comment:
```suggestion
```

Reply:
Done. Making all these changes in a new commit